common/src/files.rs

```diff
@@ -0,0 +1,134 @@
+use crate::utils::keccak;
+use crate::Span;
+use codespan_reporting as cs;
+use cs::files::Error as CsError;
+use std::{
+    collections::HashMap,
+    convert::TryInto,
+    fs,
+    io,
+    ops::Range,
+    path::Path,
+};
+
+pub struct SourceFile {
+    id: SourceFileId,
+    name: String,
+    content: String,
+    line_starts: Vec<usize>,
+}
+
+#[derive(PartialEq, Copy, Clone, Eq, Hash)]
+pub struct SourceFileId(u128);
```

Comment:
No great reason. I thought it would be reasonable to use content hash as the file id (though I'm not convinced that's ideal; rustc just uses file path I think), keccak256 is available in the same crate, u128 is the biggest built-in number, so I figured I'd just take half a keccak and stick it in the u128 and call it good :)